service/controller/resource/ciliumnetpol/resource.go

```diff
@@ -46,6 +48,43 @@ func toCiliumNetworkPolicy(v interface{}) (*unstructured.Unstructured, error) {
 		return nil, microerror.Mask(err)
 	}
 
+	worldPorts := []map[string]string{
+		{
+			"port": "443",
+		},
+		// Grafana cloud mimir port
+		{
+			"port": "6443",
+		},
+		// Grafana cloud squid proxy port
+		{
+			"port": "3128",
+		},
+	}
+	// We need to retrieve the proxy port from the environment variables
+	// and add it to the CiliumNetworkPolicy.
+	proxyString, ok := os.LookupEnv("HTTP_PROXY")
+	if !ok {
+		proxyString, ok = os.LookupEnv("HTTPS_PROXY")
+		if !ok {
+			proxyString, ok = os.LookupEnv("http_proxy")
+			if !ok {
+				proxyString, ok = os.LookupEnv("https_proxy")
+			}
+		}
+	}
+	if ok {
+		proxyURL, err := url.Parse(proxyString)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		proxyPort := proxyURL.Port()
+		if proxyPort == "" {
+			proxyPort = "80"
```

Comment:
Well this is nasty but if it works

Reply:
Do you have a suggestion to avoid that ?

Reply:
Well, the think is if it is https, it should be 443 and not 80 but no the nasty part was more about the proxy things above btw

Reply:
Code refactored, tell me ;-)

Reply:
Nice :)